src/main/java/de/tum/cit/aet/artemis/atlas/web/LearnerProfileResource.java

```diff
@@ -52,19 +44,19 @@ public LearnerProfileResource(UserRepository userRepository, CourseLearnerProfil
     }
 
     /**
-     * GET /learner-profiles/course-learner-profiles : get a Map of a {@link de.tum.cit.aet.artemis.core.domain.Course} id
+     * GET course-learner-profiles : get a Map of a {@link de.tum.cit.aet.artemis.core.domain.Course} id
      * to the corresponding {@link CourseLearnerProfile} of the logged-in user.
      *
      * @return The ResponseEntity with status 200 (OK) and with the body containing a map of DTOs, which contains per course profile data.
      */
     @GetMapping("course-learner-profiles")
     @EnforceAtLeastStudent
-    public ResponseEntity<Map<Long, CourseLearnerProfileDTO>> getCourseLearnerProfiles() {
+    public ResponseEntity<Set<CourseLearnerProfileDTO>> getCourseLearnerProfiles() {
         User user = userRepository.getUser();
         log.debug("REST request to get all CourseLearnerProfiles of user {}", user.getLogin());
-        Map<Long, CourseLearnerProfileDTO> result = courseLearnerProfileRepository.findAllByLogin(user.getLogin()).stream()
-                .collect(Collectors.toMap(profile -> profile.getCourse().getId(), CourseLearnerProfileDTO::of));
-        return ResponseEntity.ok(result);
+        Set<CourseLearnerProfileDTO> courseLearnerProfiles = courseLearnerProfileRepository.findAllByLogin(user.getLogin()).stream().map(CourseLearnerProfileDTO::of)
+                .collect(Collectors.toSet());
+        return ResponseEntity.ok(courseLearnerProfiles);
     }
 
     /**
```

Comment:
The javadoc in line 63 broke

Would need to be 
`Validates that fields are within {@link CourseLearnerProfile#MIN_PROFILE_VALUE} and {@link CourseLearnerProfile#MAX_PROFILE_VALUE}.` instead